### PR TITLE
Beta cleanup2

### PR DESCRIPTION
**Pull request by PhilPJL**
Started changing List<T> to IEnumerable<T> and params.
This allows us to refactor TinkerSpellList for example.

### CODE REVIEW

File: SolastaCommunityExpansion/Classes/Tinkerer/TinkererSpellList.cs [hunk 1]
```diff
@@ -11,108 +13,23 @@ private TinkererSpellList(string name, string guid, GuiPresentation guiPresentat
...
-            cantrips.Spells.Add(DatabaseHelper.SpellDefinitions.DancingLights);
-            cantrips.Spells.Add(DatabaseHelper.SpellDefinitions.FireBolt);
-            cantrips.Spells.Add(DatabaseHelper.SpellDefinitions.Guidance);
-            cantrips.Spells.Add(DatabaseHelper.SpellDefinitions.Light);
-            cantrips.Spells.Add(DatabaseHelper.SpellDefinitions.PoisonSpray);
-            cantrips.Spells.Add(DatabaseHelper.SpellDefinitions.RayOfFrost);
-            cantrips.Spells.Add(DatabaseHelper.SpellDefinitions.Resistance);
-            cantrips.Spells.Add(DatabaseHelper.SpellDefinitions.ShockingGrasp);
-            cantrips.Spells.Add(DatabaseHelper.SpellDefinitions.SpareTheDying);
-            // melee weapon attack (booming blade/green flame blade)
```
Comment (ChrisPJohn) [inline SolastaCommunityExpansion/Classes/Tinkerer/TinkererSpellList.cs:28]: A lot of the comments here were about spells to be added (either if they were included in the game or as additional content in a mod. I'd like to keep that documented somewhere, either in comments here still or as issues.

File: SolastaCommunityExpansion/Classes/Tinkerer/FeatureHelpers.cs [hunk 2]
```diff
@@ -350,75 +359,88 @@ public class FeatureDefinitionAdditionalDamageBuilder : BaseDefinitionBuilder<Fe
                 Definition.SetAdditionalDamageType(additionalDamageType);
                 Definition.SetSpecificDamageType(specificDamageType);
                 Definition.SetDamageAdvancement(damageAdvancement);
-                Traverse.Create(Definition).Field("diceByRankTable").SetValue(diceByRankTable);
+                Definition.DiceByRankTable.SetRange(diceByRankTable);
```
Comment (ChrisPJohn) [inline SolastaCommunityExpansion/Classes/Tinkerer/FeatureHelpers.cs:362]: I don't think SetRange is guaranteed to have the same behavior as SetValue. This should make sure to fully replace the existing DiceByRankTable.

File: SolastaCommunityExpansion/Classes/Tinkerer/FeatureHelpers.cs [hunk 3]
```diff
@@ -535,11 +547,11 @@ public static DiceByRank BuildDiceByRank(int rank, int dice)
 
         public class FeatureDefinitionFeatureSetBuilder : BaseDefinitionBuilder<FeatureDefinitionFeatureSet>
         {
-            public FeatureDefinitionFeatureSetBuilder(string name, string guid, List<FeatureDefinition> featureSet,
+            public FeatureDefinitionFeatureSetBuilder(string name, string guid, IEnumerable<FeatureDefinition> featureSet,
                 FeatureDefinitionFeatureSet.FeatureSetMode mode, int defaultSelection, bool uniqueChoices,
                 bool enumerateInDescription, GuiPresentation guiPresentation) : base(name, guid)
             {
-                Traverse.Create(Definition).Field("featureSet").SetValue(featureSet);
+                Definition.SetField("featureSet", featureSet.ToList());
```
Comment (ChrisPJohn) [inline SolastaCommunityExpansion/Classes/Tinkerer/FeatureHelpers.cs:554]: Could you highlight the manual changes in this file? I assume this line was manual was, but most of this automated.

Most of this file can probably be just replaced by calls to feature builders already in CE, so I don't particularly want to spend a lot of time reviewing the file.

**Review by ChrisPJohn — CHANGES_REQUESTED** [review]

File: SolastaCommunityExpansion/Classes/Tinkerer/TinkererSpellList.cs (hunk 1 above)
Comment (PhilPJL) [reply SolastaCommunityExpansion/Classes/Tinkerer/TinkererSpellList.cs:28]: done

File: SolastaCommunityExpansion/Classes/Tinkerer/FeatureHelpers.cs (hunk 2 above)
Comment (PhilPJL) [reply SolastaCommunityExpansion/Classes/Tinkerer/FeatureHelpers.cs:362]: I did check, this field is being initialized. 

```
    [SerializeField]
    [Tooltip("Determines how the number of dice for each rank defined by <damageAdvancement>")]
    private List<DiceByRank> diceByRankTable = new List<DiceByRank>();

```

and SetRange does
```
        public static void SetRange<T>(this List<T> list, IEnumerable<T> range)
        {
            list.Clear();
            list.AddRange(range);
        }

```

File: SolastaCommunityExpansion/Classes/Tinkerer/FeatureHelpers.cs (hunk 3 above)
Comment (PhilPJL) [reply SolastaCommunityExpansion/Classes/Tinkerer/FeatureHelpers.cs:554]: Most of the changes are manual but some of them I used the change signature refactoring to make the collection parameter the last parameter. Some of them I've used refactorings individually like inline local variable.

**Review by ChrisPJohn — APPROVED** [review]

**Pull request merged**